src/cli/msig/index.ts

```diff
@@ -163,6 +166,75 @@ export default async function msig(args: string[]) {
                 newGovernanceAccounts
             );
 
+            if (test) await executeProposal(proposer, proposalName, proposalHash);
+        } else if (proposalType === 'deploy-contract') {
+            const contractName = args[3];
+
+            if (!contractName) {
+                throw new Error('Contract name must be provided for deploy-contract proposal');
+            }
+
+            const contractInfo = {
+                account: contractName,
+                contractDir: path.join(__dirname, `../../Tonomy-Contracts/contracts/${contractName}`),
+            };
+
+            const action = {
+                account: 'eosio',
+                name: 'setcode',
+                authorization: [
+                    {
+                        actor: contractName,
+                        permission: 'active',
+                    },
+                ],
+                data: {
+                    account: contractName,
+                    vmtype: 0,
+                    vmversion: 0,
+                    code: await deployContract(contractInfo, signer, { throughTonomyProxy: true }),
+                },
+            };
+
+            const proposalHash = await createProposal(
+                proposer,
+                proposalName,
+                [action],
+                privateKey,
+                newGovernanceAccounts
+            );
+
+            if (test) await executeProposal(proposer, proposalName, proposalHash);
+        } else if (proposalType === 'eosio.code') {
+            const contractAccount = 'vesting.tmy';
+            const permission = 'ops.tmy';
+
+            const action = {
+                account: 'eosio',
+                name: 'setcode',
```

Comment:
look at the tonomyContract.updateAuth() action call and what goes into it. Also check the CLI at how the `authority` value is created for each of the allocated funds pools